### ISSUE TITLE
Zed Flatpak: Missing DSO in Linux archive binaries

### ISSUE THREAD

**Issue by jerome-benoit** [zed-industries/zed#18450]
### Check for existing issues

- [X] Completed

### Describe the bug / provide steps to reproduce it

zed flatpak issue is triggered: 

- `flatpak: /var/lib/flatpak/app/dev.zed.Zed/x86_64/stable/43109795a0a1489151b14d6bc71ebb7b3117a420fd59353f339ac8a28f1fdc75/files/lib/libssl.so.3: version 'OPENSSL_3.2.0' not found (required by /lib64/libcurl.so.4)`

zed is trying to use the host libcurl but with the sandboxed libssl shipped in the flatpak and they are ABI incompatible.  

It causes several failures at zed flatpack usage when `remote-server` download is needed. 

Bundling curl into the flatpak against the shipped libssl is not possible: the headers, ... needed are not shipped in the archive, see https://github.com/flathub/dev.zed.Zed/pull/26

### Environment

Not relevant. 

### If applicable, attach your Zed.log file to this issue.

<details><summary>Zed.log</summary><pre>

</pre></details>

**Comment by dangrystad** [comment]
I have the same problem on flatpak on Fedora 40
```
$ flatpak info dev.zed.Zed

Zed - High-performance code editor

          ID: dev.zed.Zed
         Ref: app/dev.zed.Zed/x86_64/stable
        Arch: x86_64
      Branch: stable
     Version: v0.154.3
     License: AGPL-3.0-or-later and Apache-2.0 and GPL-3.0-or-later
      Origin: flathub
  Collection: org.flathub.Stable
Installation: system
   Installed: 156,7 MB
     Runtime: org.freedesktop.Sdk/x86_64/24.08
         Sdk: org.freedesktop.Sdk/x86_64/24.08

      Commit: d0dfc5e66c1a8cfcaab3527a52e847851783264882dee9a509b7ec82aee9a801
      Parent: 2f4c20a9b00c9b0de1f725ad309b69c3ca94354ec9343e4d7003b02177d90833
     Subject: Update to v0.154.3 (#27) (e1c1779e)
        Date: 2024-09-27 23:59:28 +0000
```

**Comment by ernanvr** [comment]
Same issue
![imagen](https://github.com/user-attachments/assets/6d43d7ea-98cc-4771-a2a9-1c9cbe5a2e17)

**Comment by notpeter** [comment]
Closing because @jerome-benoit confirmed in this thread this was fixed in 0.155.x:
- https://github.com/flathub/dev.zed.Zed/issues/21#issuecomment-2397928943
